src/Migration/Cache.php

```diff
@@ -75,14 +80,23 @@ public function addAll(array $resources): void
      */
     public function update(Resource $resource): void
     {
-        if (! in_array($resource->getName(), $this->cache)) {
-            $this->add($resource);
-        }
-
+        // if documents then updating the status counter only
         if ($resource->getName() == Resource::TYPE_DOCUMENT) {
+            $status = $resource->getStatus();
+            $name = $resource->getName();
+            if (!isset($this->cache[$name][$status])) {
+                $this->add($resource);
+            }
+            if (is_int($this->cache[$resource->getName()][$status])) {
+                $this->cache[$resource->getName()][$status]++;
+            }
             return;
         }
 
+        if (! in_array($resource->getName(), $this->cache)) {
+            $this->add($resource);
+        }
+
```

Comment:
If a document status changes, we should decrement the old status and increment the new - we need to know what the old status is per-document